Weather_Display_Matrix/openweather_graphics.py

```diff
@@ -0,0 +1,205 @@
+
+        self.city_text = None
+
+        self.temp_text = Label(self.medium_font, max_glyphs=6)
+        self.temp_text.x = 20
+        self.temp_text.y = 7
+        self.temp_text.color = TEMP_COLOR
+        self._text_group.append(self.temp_text)
+
+        self.description_text = Label(self.small_font, max_glyphs=60)
+        self.description_text.color = DESCRIPTION_COLOR
+        self._scrolling_texts.append(self.description_text)
+
+        self.humidity_text = Label(self.small_font, max_glyphs=14)
+        self.humidity_text.color = HUMIDITY_COLOR  #
+        self._scrolling_texts.append(self.humidity_text)
+
+        self.wind_text = Label(self.small_font, max_glyphs=10)
+        self.wind_text.color = WIND_COLOR
+        self._scrolling_texts.append(self.wind_text)
+
+    def display_weather(self, weather):
+        # set the icon
+        self.set_icon(weather["weather"][0]["icon"])
+
+        city_name = weather["name"] + ", " + weather["sys"]["country"]
+        print(city_name)
+        if not self.city_text:
+            self.city_text = Label(self.small_font, text=city_name)
+            self.city_text.color = CITY_COLOR
+            self._scrolling_texts.append(self.city_text)
+
+        temperature = weather["main"]["temp"]
+        print(temperature)
+        if self.celsius:
+            self.temp_text.text = "%d°C" % temperature
+        else:
+            self.temp_text.text = "%d°F" % temperature
+
+        description = weather["weather"][0]["description"]
+        description = description[0].upper() + description[1:]
+        print(description)
+        self.description_text.text = description
+        # "thunderstorm with heavy drizzle"
+
+        humidity = weather["main"]["humidity"]
+        print(humidity)
+        self.humidity_text.text = "%d%% humidity" % humidity
+
+        wind = weather["wind"]["speed"]
+        print(wind)
+        if self.meters_speed:
+            self.wind_text.text = "%d m/s" % wind
+        else:
+            self.wind_text.text = "%d mph" % wind
+
+        self.display.show(self.root_group)
+
+    def set_icon(self, icon_name):
+        """Use icon_name to get the position of the sprite and update
+        the current icon.
+
+        :param icon_name: The icon name returned by openweathermap
+
+        Format is always 2 numbers followed by 'd' or 'n' as the 3rd character
+        """
+
+        icon_map = ("01", "02", "03", "04", "09", "10", "11", "13", "50")
+
+        print("Set icon to", icon_name)
+        if self._icon_group:
+            self._icon_group.pop()
+        if icon_name is not None:
+            row = None
+            for index, icon in enumerate(icon_map):
+                if icon == icon_name[0:2]:
+                    row = index
+                    break
+            column = 0
+            if icon_name[2] == "n":
+                column = 1
+            if row is not None:
+                print(column, row)
```

Comment:
Maybe you can remove this print statement. I accidentally left it in for debugging and forgot to remove it.

Reply:
removed!